### ISSUE TITLE
Inconsistencies with the published `1.0.0-alpha.1`(crates.io) versus `v1.0.0-alpha.1` (GitHub)

### ISSUE THREAD

**Issue by realeinherjar** [bitcoindevkit/bdk#1059]
If you try to run the `wallet_esplora_async` example from the `v1.0.0-alpha.1` tag:
https://github.com/bitcoindevkit/bdk/blob/f4d2a76661d465dd0640d54470b95cc7b5ece979/example-crates/wallet_esplora_async/src/main.rs#L1-L97

you'll need to set your `Cargo.toml` dependencies to the `v1.0.0-alpha.1` tag:

```toml
[dependencies]
bdk = { git = "https://github.com/bitcoindevkit/bdk", tag = "v1.0.0-alpha.1" }
bdk_esplora = { git = "https://github.com/bitcoindevkit/bdk", tag = "v1.0.0-alpha.1", features = [
  "async-https",
] }
bdk_chain = { git = "https://github.com/bitcoindevkit/bdk", tag = "v1.0.0-alpha.1", features = [
  "serde",
] }
bdk_file_store = { git = "https://github.com/bitcoindevkit/bdk", tag = "v1.0.0-alpha.1" }
tokio = { version = "1", features = ["rt", "rt-multi-thread", "macros"] }
```

and it runs as expected.

However, if you try to run with the published `bdk@1.0.0-alpha.1` from crates.io with `cargo add bdk@1.0.0-alpha.1`, you'll get this error:

```rust
error[E0599]: the function or associated item `new_from_path` exists for struct `Store<'_, LocalChangeSet<KeychainKind, ConfirmationTimeAnchor>>`, but its trait bounds were not satisfied
   --> src/main.rs:20:47
    |
20  |     let db = Store::<bdk::wallet::ChangeSet>::new_from_path(DB_MAGIC.as_bytes(), db_path)?;
    |                                               ^^^^^^^^^^^^^ function or associated item cannot be called due to unsatisfied trait bounds
    |
   ::: /Users/storopoli/.cargo/registry/src/index.crates.io-6f17d22bba15001f/bdk_chain-0.5.0/src/keychain.rs:127:1
    |
127 | pub struct LocalChangeSet<K, A> {
    | ------------------------------- doesn't satisfy `_: Append`
    |
    = note: the following trait bounds were not satisfied:
            `bdk::bdk_chain::keychain::LocalChangeSet<KeychainKind, bdk::bdk_chain::ConfirmationTimeAnchor>: bdk_chain::Append`

For more information about this error, try `rustc --explain E0599`.
```

**Comment by notmandatory** [comment]
Which version of rust are you using? did you do a `cargo update` first? I tried running the `wallet_esplora_async` example from the `v1.0.0-alpha.1` tag and from the current latest `master` branch version (`feafaaca31a0a40afc03ce98591d151c48c74fa2`) without any changes and I'm not getting the same error you got. 

I did get a runtime error that I think is related to `blocksteam.info` rate limiting or some other issue on their side. When I switched the example to `mempool.space` the example works fine:

```rust
let client =
        esplora_client::Builder::new("https://mempool.space/testnet/api").build_async()?;
```
